Doc/library/socket.rst

```diff
@@ -225,6 +225,33 @@ created.  Socket addresses are represented as follows:
 
   .. versionadded:: 3.9
 
+- :const:`AF_HYPERV` is a Windows-only socket based interface for communicating
+  with Hyper-V hosts and guests. The address family is represented as a
+  ``(vm_id, service_id)`` tuple where the ``vm_id`` and ``service_id`` are the
+  little endian byte representation of a ``uuid.UUID`` object.
```

Comment:
Would it be more user-friendly for the connect API to receive two strings and then let the C code do the LE conversion?

Reply:
I think it would be but this is where my lack of C abilities fall short. I'm happy to pursue it if you think it's a better way forward but I wouldn't know how to do this nicely in the C side.

Reply:
IMO, that makes for a better Python API, so that's what I would recommend FWIW. Can you use the Windows API to create an UUID from a string?[^1]?


[^1]: I'm don't know much about Windows, but I found a couple of candidates while browsing the Windows C/C++ API docs; perhaps `UuidFromString`?

Reply:
It's certainly possible, probably simple to add so I'll look into it. I wasn't sure if there was some way in Python to do this already through the `uuid.UUID` class but calling `UuidFromString` (or another equivalent Win32 API) should be simple to do.

Reply:
> I wasn't sure if there was some way in Python to do this already through the uuid.UUID class [...]

Sure, here's an example from the sqlite3 module, where we load `functools.lru_cache` during module import, and then call it during connection init:

https://github.com/python/cpython/blob/b7f83bdd0e16f82288dc3557b7e715bb5c7d96d0/Modules/_sqlite/module.c#L258-L273

https://github.com/python/cpython/blob/b7f83bdd0e16f82288dc3557b7e715bb5c7d96d0/Modules/_sqlite/connection.c#L138-L159

Reply:
If bytes are already accepted here, I think it's fine to just require them and offer the recipe (and perhaps some constants).

Reply:
I went ahead and converted the code to accept strings rather than bytes as I felt it was more friendly to being calling the function. It also means I could easily add constants for those known addr UUIDs by using `PyModule_AddStringConstant`.